onnxruntime/core/framework/op_kernel.cc

```diff
@@ -218,4 +221,147 @@ Status OpKernelContext::SetOutputMLValue(int index, const OrtValue& ort_value) {
 }
 #endif
 
+// InstantKernelContext
+InstantKernelContext::InstantKernelContext(const OrtValue* const* input_values,
+                                           int input_count,
+                                           OrtValue* const* output_values,
+                                           int output_count,
+                                           AllocatorPtr allocator,
+                                           onnxruntime::concurrency::ThreadPool* threadpool,
+                                           const logging::Logger& logger) : OpKernelContext(threadpool, logger),
+                                                                            input_values_(input_values),
+                                                                            input_count_(input_count),
+                                                                            output_values_(output_values),
+                                                                            output_count_(output_count),
+                                                                            allocator_(allocator) {}
+
+int InstantKernelContext::NumVariadicInputs(size_t arg_num) const {
+  auto ort_value = input_values_[arg_num];
```

Comment:
>input_values_ [](http://example.com/codeflow?start=19&length=13)

todo: we should check the validity of "arg_num" and throw a friendly exception when necessary.